lib/scheduler.py

```diff
@@ -47,10 +47,12 @@ def _dfs(self, packages, visited):
         else:
             if p not in visited:
                 visited.append(p)
-                if p.dependencies:
-                    order.extend(self._dfs(p.dependencies, visited))
                 if p.build_dependencies:
                     order.extend(self._dfs(p.build_dependencies, visited))
                 order.append(p)
+                # runtime dependencies do not need to be built before the package,
+                # they can be built anytime
+                if p.dependencies:
+                    order.extend(self._dfs(p.dependencies, visited))
```

Comment:
If they can be built at any time, there's no need to deliberately place them after the dependent package.
One reason it could be useful to have the install dependency built before is that it could be installed as soon as the RPM is built, while the dependent package is still building. The opposite could not be done. 
Also, if the dependency build fails, we'd end with a dependent RPM that could not be installed, while, with the opposite, if any package build fails, all the ones that succeeded are installable.